_posts/2024-08-24-GSoC2024-dingo.md

```diff
@@ -0,0 +1,230 @@
+
+In the future, other clustering methods such as `k-means`, `HDBSCAN`, and `biclustering` at the sample level will be explored to identify more specific clusters. Let me briefly explain: Some pathways have distinct phases, each serving a unique role in metabolism. Both glycolysis and the pentose phosphate pathways can be divided into two phases.
+
+For example, the pentose phosphate pathway has an oxidative and a non-oxidative phase. Reactions such as `G6PDH2R`, `PGL`, and `GND` belong to the oxidative phase, which produces `NADPH`. Hierarchical clustering may not distinguish these phases, but using alternative methods not solely based on the correlation matrix could reveal two distinct clusters representing the two phases.
+
+
+## Graphs from the correlation matrix
+
+- Creating graphs can reveal networks of correlated reactions, which may correspond to metabolic pathways.
+- These graphs are generated from the correlation matrix, where reactions are represented as nodes and correlation coefficients as edges.
+
+The `graph_corr_matrix` function generates graphs from the correlation matrix, and the `plot_graph` function plots these graphs.
+
+Code that illustrates how to use these functions:
+```python
+graphs, layouts = graph_corr_matrix(corr_matrix,
+                                    reactions,
+                                    # boolean value; if True, transforms the correlation matrix values into absolute values
+                                    correction = True,
+                                    # nested list containing sublists of reactions within the same cluster, created by the `cluster_corr_reactions` function.
+                                    clusters = clusters)
+                                
+plot_graph(
+           # graph object returned by the graph_corr_matrix function
+           graph,
+           # layout corresponding to the given graph object 
+           layout)
+
+```
+Here’s an example of a graph created from a correlation matrix without cutoffs:
+<center>
+    <img src="https://sotiristouliopoulos.github.io/dingo/img/graph_no_cutoffs.png"><br><br>
+</center>
+
+The width of each edge represents the correlation strength, scaled from 0 to 1. In addition to the initial graph, users can create and plot subgraphs that represent reaction subnetworks. These subnetworks can serve as supplementary information to clustering results, allowing researchers to identify similarities and differences between the two methods and adjust the structure of desired clusters accordingly.
+
+However, the insights gained from graphs extend beyond pathway prediction. The more connected a node (reaction) is, the more essential its role likely is in metabolism. Conversely, nodes with fewer connections may represent reactions that can be removed without significantly affecting the system.
```

Comment:
Could we give two examples from the graph in the figure, one essential node and one that is not significant and if those examples make sense biologically.